app/src/test/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepositoryTest.kt

```diff
@@ -0,0 +1,130 @@
+
+        private val LECTURE_1001 = Lecture("1001").apply {
+            changedIsCanceled = false
+            changedTitle = false
+            changedIsNew = false
+            url = "" // only initialized for toLecturesDatabaseModel()
+        }
+
+        private val LECTURE_1002 = Lecture("1002").apply {
+            changedIsCanceled = true
+            changedTitle = false
+            changedIsNew = false
+            url = "" // only initialized for toLecturesDatabaseModel()
+        }
+
+        private val LECTURE_1003 = Lecture("1003").apply {
+            changedIsCanceled = false
+            changedTitle = true
+            changedIsNew = false
+            url = "" // only initialized for toLecturesDatabaseModel()
+        }
+
+        private val LECTURE_1004 = Lecture("1004").apply {
+            changedIsCanceled = false
+            changedTitle = false
+            changedIsNew = true
+            url = "" // only initialized for toLecturesDatabaseModel()
+        }
+
+        private val LECTURE_1005 = Lecture("1005").apply {
+            changedIsCanceled = true
+            changedTitle = true
+            changedIsNew = true
+            url = "" // only initialized for toLecturesDatabaseModel()
+        }
+
+        private val LECTURE_2001 = Lecture("2001").apply {
+            highlight = false
+            changedIsCanceled = false
+            url = "" // only initialized for toLecturesDatabaseModel()
+        }
+
+        private val LECTURE_2002 = Lecture("2002").apply {
+            highlight = true
+            changedIsCanceled = false
+            url = "" // only initialized for toLecturesDatabaseModel()
+        }
+
+        private val LECTURE_2003 = Lecture("2003").apply {
+            highlight = true
+            changedIsCanceled = true
+            url = "" // only initialized for toLecturesDatabaseModel()
+        }
+
+        private val LECTURE_2004 = Lecture("2004").apply {
+            highlight = false
+            changedIsCanceled = true
+            url = "" // only initialized for toLecturesDatabaseModel()
+        }
+
+    }
+
+    @Test
+    fun loadChangedLecturesWithEmptyList() {
+        whenever(lecturesDatabaseRepository.queryLecturesOrderedByDateUtc()) doReturn emptyList()
+        assertThat(testableAppRepository.loadChangedLectures()).isEmpty()
+    }
+
+    @Test
+    fun loadChangedLecturesWithAllLectures() {
+        val lectures = listOf(LECTURE_1001, LECTURE_1002, LECTURE_1003, LECTURE_1004, LECTURE_1005)
+        whenever(lecturesDatabaseRepository.queryLecturesOrderedByDateUtc()) doReturn lectures.toLecturesDatabaseModel()
+        val changedLectures = testableAppRepository.loadChangedLectures()
+        assertThat(changedLectures).isNotEmpty()
+        assertThat(changedLectures.size).isEqualTo(4)
+        assertThat(changedLectures).containsExactly(LECTURE_1002, LECTURE_1003, LECTURE_1004, LECTURE_1005)
+        assertThat(changedLectures).doesNotContain(LECTURE_1001)
```

Comment:
Yes, the test is very explicit aka. redundant in that respect. I was aware of it but left it since it wasn't harming anyone. I will thin out the assertions.

Reply:
I simplified the test(s) in 97986cc61bf7a10b3c7403a33ae42990731c7fd8 and the following commits.